dbms/src/Storages/StorageKafka.cpp

```diff
@@ -187,19 +188,20 @@ StorageKafka::StorageKafka(
     columns(columns_), topics(topics_), format_name(format_name_), schema_name(schema_name_),
     conf(rd_kafka_conf_new()), log(&Logger::get("StorageKafka (" + table_name_ + ")"))
 {
-    char errstr[512];
+    std::vector<char> errstr;
+    errstr.reserve(512);
```

Comment:
BTW, 512 bytes on the stack is totally fine (if the function is not called with non-tail recursion).